src/button/Button.tsx

```diff
@@ -0,0 +1,121 @@
+import "./_button.scss";
+
+import React, {useEffect, useRef} from "react";
+import classNames from "classnames";
+
+import Spinner from "../spinner/Spinner";
+
+export interface ButtonProps {
+  testid: string;
+  children: React.ReactNode;
+  onClick?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseOver?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseDown?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseUp?: React.ReactEventHandler<HTMLButtonElement>;
+  onFocus?: React.ReactEventHandler<HTMLButtonElement>;
+  onBlur?: React.ReactEventHandler<HTMLButtonElement>;
+  id?: string;
+  lang?: string;
+  type?: "button" | "submit" | "reset";
+  shouldDisplaySpinner?: boolean;
+  isDisabled?: boolean;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  shouldFocus?: boolean;
+  ariaLabel?: string;
+  customClassName?: string;
+  tabIndex?: number;
+}
+
+function Button({
+  testid,
+  type = "button",
+  id,
+  lang,
+  onClick,
+  children,
+  customClassName,
+  shouldPreventDefault = true,
+  shouldStopPropagation = true,
+  shouldFocus,
+  shouldDisplaySpinner,
+  ariaLabel,
+  isDisabled,
+  onMouseOver,
+  onMouseDown,
+  onMouseUp,
+  onFocus,
+  onBlur,
+  tabIndex
+}: ButtonProps) {
+  const shouldInactivate = Boolean(isDisabled || shouldDisplaySpinner);
```

Comment:
The name sounds like it causes a side effect to happen. I think `isButtonInactive` or `isButtonDisabled` is a more suited name for this variable. What do you think ?

Reply:
👍 Going with `isButtonDisabled`. Simple and straight to the point